### PR TITLE
Feature/dependency injection

### PR DESCRIPTION
**Pull request by JSchraff**

### CODE REVIEW

**Comment by Death111** [comment]
looks good so far. Merge conflicts have to be resolved and I'll have a look again.

File: src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java [hunk 1]
```diff
@@ -115,6 +107,15 @@ private void initialize() {
 
       Interval.registerCallBack(() -> projectsListViewController.tick());
 
+      projectsListViewController = new ProjectsListViewController(model, controller, stage, projectListView,
```
Comment (Death111) [inline src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java]: stage is null here, resulting in npe with popupstage when "change with time" is used.
we may still need the 2nd initialize here?

File: src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java [hunk 2]
```diff
@@ -115,6 +107,15 @@ private void initialize() {
 
       Interval.registerCallBack(() -> projectsListViewController.tick());
```
Comment (Death111) [inline src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java]: if we look at it like this, this seems dangerous, registering callbacks on an object, which is instantiated one line after the registration :)

File: src/main/java/de/doubleslash/keeptime/view/ReportController.java [hunk 3]
```diff
@@ -101,22 +104,31 @@
...
+
    @FXML
    private void initialize() {
       LOG.info("Init reportController");
       currentReportDate = LocalDate.now();
-      initTableView();
       colorTimeLine = new ColorTimeLine(colorTimeLineCanvas);
+
+      loadCalenderWidget();
+      initTableView();
```
Comment (Death111) [inline src/main/java/de/doubleslash/keeptime/view/ReportController.java]: can you please have a look if I merged this correctly? It is working :)

File: src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java (hunk 2 above)
Comment (JSchraff) [reply src/main/java/de/doubleslash/keeptime/viewpopup/ViewControllerPopup.java]: yes it is merged correctly

**Pull request merged**